File: PUGD-api/models/AcquisitionModels/Order.go
package AcquisitionModels

import (
	"go.mongodb.org/mongo-driver/bson/primitive"
)

type Order struct {
	ID               primitive.ObjectID   `json:"_id" bson:"_id"`
	Type             string               `json:"type"`
	Name             string               `json:"name"`
	Quotation_number string               `json:"quotation_number"`
	Order_number     string               `json:"order_number"`
	Status           string               `json:"status"`
	Establishement   string               `json:"establishement"`
	Provider         string               `json:"provider"`
	Financial_year   string               `json:"financial_year"`
	Currency         string               `json:"currency"`
	Date             string               `json:"date"` //to convert to date
	Delivery_address string               `json:"delivery_address"`
	Billing_address  string               `json:"billing_address"`
	Notes            string               `json:"notes"`
	Order_lines      []primitive.ObjectID `json:"order_lines"`
	Orders           []primitive.ObjectID `json:"orders"`
	Received         bool                 `json:received`
	Factured         bool                 `json:factured`
}

File: PUGD-frontend/graphql/mutations/acquisition/order.js
import gql from "graphql-tag";

const InsertOrder = gql`
  mutation(
    $id: String!
    $establishement: String!
    $name: String
    $quotation_number: String
    $order_number: String
    $financial_year: String!
    $date: String!
    $delivery_address: String!
    $billing_address: String!
    $notes: String!
    $status: String!
    $type: String!
    $provider: String!
    $order_lines: [String!]!
  ) {
    insertOrder(
      id: $id
      establishement: $establishement
      name: $name
      quotation_number: $quotation_number
      order_number: $order_number
      financial_year: $financial_year
      date: $date
      delivery_address: $delivery_address
      billing_address: $billing_address
      notes: $notes
      status: $status
      type: $type
      provider: $provider
      order_lines: $order_lines
    )
  }
`;

const UpdateOrder = gql`
  mutation(
    $_id: String!
    $establishement: String
    $name: String
    $quotation_number: String
    $order_number: String
    $financial_year: String
    $date: String
    $delivery_address: String
    $billing_address: String
    $notes: String
    $status: String
    $type: String
    $provider: String
    $order_lines: [String]
  ) {
    updateOrder(
      _id: $_id
      establishement: $establishement
      name: $name
      quotation_number: $quotation_number
      order_number: $order_number
      financial_year: $financial_year
      date: $date
      delivery_address: $delivery_address
      billing_address: $billing_address
      notes: $notes
      status: $status
      type: $type
      provider: $provider
      order_lines: $order_lines
    )
  }
`;

const DeleteOrder = gql`
  mutation($_id: String!) {
    deleteOrder(_id: $_id)
  }
`;

//important
module.exports = {
  InsertOrder,
  DeleteOrder,
  UpdateOrder,
};
